src/imitation/scripts/analyze.py

```diff
@@ -85,6 +145,32 @@ def _make_return_summary(stats: dict, prefix="") -> str:
     stats["n_traj"])
 
 
+def _get_sacred_dicts(source_dir: str,
+                      run_name: str,
+                      env_name: str,
+                      skip_failed_runs: bool) -> List[sacred_util.SacredDicts]:
+  sacred_dirs = sacred_util.filter_subdirs(source_dir)
+  sacred_dicts = [sacred_util.SacredDicts.load_from_dir(sacred_dir)
+                  for sacred_dir in sacred_dirs]
+
+  if run_name is not None:
```

Comment:
Consider factoring out lines 156-169 into a helper method. The logic seems very similar to that of lines 96-109.

Reply:
Oops, I should have replaced 96-109 with a call to this function. Doing that now